src/extractor/guidance/turn_handler.cpp

```diff
@@ -553,9 +521,18 @@ std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
             return true;
         }();
 
+        const bool has_invalid_entry = [&]() {
+            const auto end_itr = intersection.begin() + left + 1;
+            return end_itr !=
+                   std::find_if(intersection.begin() + right,
```

Comment:
Actually seeing that you use `!has_invalid_entry` why not resolve the double negation and and implement `only_valid_entries` instead?